src/Castle.Windsor/MicroKernel/Handlers/DefaultGenericHandler.cs

```diff
@@ -364,7 +364,7 @@ private IDictionary GetExtendedProperties()
 			var extendedProperties = ComponentModel.ExtendedProperties;
 			if (extendedProperties != null && extendedProperties.Count > 0)
 			{
-#if !SILVERLIGHT
+#if FEATURE_ICLONEABLE
 				if (extendedProperties is ICloneable)
 				{
 					extendedProperties = (IDictionary)((ICloneable)extendedProperties).Clone();
```

Comment:
@jonorossi I made now Clone() internal and kept it because otherwise that comparer will be lost and I don't know what is the effect of that.. Do you want to remove it all? So `GetExtendedProperties()` will just return `new Arguments(extendedProperties)`